src/steps/check-network-request.ts

```diff
@@ -0,0 +1,66 @@
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class CheckNetworkRequestStep extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Check for a specific network request';
+  // tslint:disable-next-line:max-line-length
+  protected stepExpression: string = 'there should be (?<reqCount>\\d+) matching network requests? for (?<baseUrl>.+)';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.VALIDATION;
+  protected expectedFields: Field[] = [{
+    field: 'reqCount',
+    type: FieldDefinition.Type.NUMERIC,
+    description: '# of Requests',
+  }, {
+    field: 'baseUrl',
+    type: FieldDefinition.Type.URL,
+    description: 'Base URL Is',
+  }, {
+    field: 'pathContains',
+    type: FieldDefinition.Type.STRING,
+    description: 'Path Contains',
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+  }, {
+    field: 'withParameters',
+    type: FieldDefinition.Type.MAP,
+    description: 'Parameters Include',
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const reqCount = stepData.reqCount;
+    const baseUrl = stepData.baseUrl;
+    const pathContains = stepData.pathContains || '';
+    const withParameters = stepData.withParameters;
+
+    try {
+      //// This will ensure that NavigateTo was called
+      await this.client.getCurrentPageInfo('url');
+
+      await this.client.waitForNetworkIdle(10000, false);
```

Comment:
- [ ] Can probably remove this now that this is in the `getNetworkRequests()`

Reply:
Oh yeah